src/include/cephfs/ceph_ll_client.h

```diff
@@ -102,6 +102,18 @@ struct ceph_statx {
  */
 #define CEPH_REQ_FLAG_MASK		(AT_SYMLINK_NOFOLLOW|AT_NO_ATTR_SYNC)
 
+/* fallocate mode flags */
+#ifdef FALLOC_FL_KEEP_SIZE
+# define CEPH_FALLOC_FL_KEEP_SIZE FALLOC_FL_KEEP_SIZE
```

Comment:
probably we should use `CEPH_FALLOC_FL_KEEP_SIZE` in `src/client/Client.cc` instead? or shall we just
```c
#ifndef FALLOC_FL_KEEP_SIZE
#define 0x01
#endif
```

as we do something similar at line 95 in the same file.

Reply:
> probably we should use `CEPH_FALLOC_FL_KEEP_SIZE` in `src/client/Client.cc` instead? or shall we just
> 
> ```c
> #ifndef FALLOC_FL_KEEP_SIZE
> #define 0x01
> #endif
> ```
> 
> as we do something similar at line 95 in the same file.

I like that idea better. Less of a change and we can keep the "normal" definitions in the code. It would be different if we were sending these values over the wire, but we aren't so that should be fine.

Reply:
Yeah, of course, I actually _did_ replace them in Client.cc but forgot to save the file before comitting. 😄

Reply:
> I like that idea better. Less of a change and we can keep the "normal" definitions in the code. It would be different if we were sending these values over the wire, but we aren't so that should be fine.

Yeah, I see the advantage of doing that. I was just worried that by defining FALLOC_FL_KEEP_SIZE I would trigger build of other code, that actually depends on it. But I guess we are safe if we do it in a client-only header.